source/Calamari/Deployment/PackageRetention/Caching/PackageQuantityPackageCacheCleaner.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.FeatureToggles;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.PackageRetention.Model;
+
+namespace Calamari.Deployment.PackageRetention.Caching
+{
+    public class PackageQuantityPackageCacheCleaner : IRetentionAlgorithm
+    {
+        const string MachinePackageCacheRetentionQuantityOfPackagesToKeep = nameof(MachinePackageCacheRetentionQuantityOfPackagesToKeep);
+        const string MachinePackageCacheRetentionQuantityOfVersionsToKeep = nameof(MachinePackageCacheRetentionQuantityOfVersionsToKeep);
+        const string PackageUnit = "MachinePackageCacheRetentionPackageUnit";
+        const string VersionUnit = "MachinePackageCacheRetentionVersionUnit";
+        const string PackageCacheRetentionStrategy = "MachinePackageCacheRetentionStrategy";
+        const MachinePackageCacheRetentionStrategy DefaultMachinePackageCacheRetentionStrategy = MachinePackageCacheRetentionStrategy.FreeSpace;
+        const int DefaultQuantityOfPackagesToKeep = -1;
+        const int DefaultQuantityOfVersionsToKeep = 5;
+        const MachinePackageCacheRetentionUnit DefaultPackageUnit = MachinePackageCacheRetentionUnit.Items;
+        const MachinePackageCacheRetentionUnit DefaultVersionUnit = MachinePackageCacheRetentionUnit.Items;
+        readonly IVariables variables;
+        readonly ILog log;
+
+        public PackageQuantityPackageCacheCleaner(IVariables variables, ILog log)
+        {
+            this.variables = variables;
+            this.log = log;
+        }
+
+        public IEnumerable<PackageIdentity> GetPackagesToRemove(IEnumerable<JournalEntry> journalEntries)
+        {
+            var packageCacheRetentionStrategyString = variables.Get(PackageCacheRetentionStrategy);
+            var retentionStrategy = packageCacheRetentionStrategyString != null ? (MachinePackageCacheRetentionStrategy) Enum.Parse(typeof(MachinePackageCacheRetentionStrategy), packageCacheRetentionStrategyString) : DefaultMachinePackageCacheRetentionStrategy;
+            
+            if (!OctopusFeatureToggles.ConfigurablePackageCacheRetentionFeatureToggle.IsEnabled(variables) || retentionStrategy == MachinePackageCacheRetentionStrategy.FreeSpace)
```

Comment:
`retentionStrategy == MachinePackageCacheRetentionStrategy.FreeSpace`
What if there is a new retention strategy `MachinePackageCacheRetentionStrategy.Time`? This condition will pass when it shouldn't. Keep the strategies decoupled.